x/models.py

```diff
@@ -45,7 +47,7 @@ def max_values(self, observation, train=False):
     def policy(self, observation, train=False):
         raise NotImplementedError
 
-    def update(self, inputs, targets):
+    def update(self, inputs, targets, actions):
```

Comment:
what about this new "actions" could explain me real quick why are we adding this.

Also, add info to docstrings please